lib/charms/tempo_k8s/v0/tracing.py

```diff
@@ -425,8 +436,20 @@ def __init__(
 
         events = self._charm.on[self._relation_name]
         self.framework.observe(events.relation_changed, self._on_tracing_relation_changed)
+        self.framework.observe(events.relation_broken, self._on_tracing_relation_broken)
+
+    @property
+    def relation(self) -> Relation:
+        """The tracing relation associated with this endpoint.
+
+        Assumes tracing has limit 1.
+        """
+        return self._charm.model.get_relation(self._relation_name)
```

Comment:
Should we really be assuming limit 1 in a library.

Reply:
yeah maybe it's not actually a bad idea to allow sending to multiple tracing endpoints

Reply:
so I think we should keep it limit 1 because:
- I don't think it is common to have a single trace source support sending traces to multiple backends. (Traefik doesn't).
- if you really want to do that, the way to do it is to put an OpenTelemetry Collector or grafana agent or something like that in the middle, that distributes the traces to multiple backends.

Reply:
@simskij sounds reasonable?

Reply:
@lucabello does this make sense to you too?

Reply:
I think the two models you mentioned are something we might want to apply elsewhere as well; I'd say let's have a small chat with the whole team and see what approach we'd rather take :)

Reply:
I think putting the GAgent in the middle if you want multiple sinks makes sense.

Reply:
GAgent will also use this lib's provider to push traces to multiple remote COS lite deployments (including multiple tempos). 
So the requirer lib should not assume limit 1. Traefik needs to set limit 1.